contrib/codeanalysis/src/python/pants/contrib/codeanalysis/tasks/index_java.py

```diff
@@ -40,6 +42,9 @@ def prepare(cls, options, round_manager):
   @classmethod
   def register_options(cls, register):
     super(IndexJava, cls).register_options(register)
+    register('--dist', type=str, choices=['none', 'uncompressed', 'tar', 'zip', 'gztar', 'bztar'],
```

Comment:
Would it be overkill to expose the indexes as a product, and do the packaging in a separate Task?

Presumably if you've renamed this package, it's because other code analysis tools might come along which could need packaging...

Reply:
They are a product already (`kythe_entries_files`), and I wondered about this. But it does seem like overkill. For example, how would we invoke this task? If it's in the `index` goal then it will always run, so we'll still need to check the flag to see if we need to do any work. And if it's in a separate 'index-bundle' or 'index-dist' goal then that seems clunky to me, and it's one more thing the user has to remember.

Reply:
OTOH, as you say, what if we have a python indexer, a scala indexer, etc. and they all emit `kythe_entries_files` products?

Reply:
OK, done! Good suggestion.

Reply:
When in doubt - products.